include/operations/StandardOperation.hpp

```diff
@@ -72,6 +72,26 @@ class StandardOperation : public Operation {
 
   [[nodiscard]] bool isStandardOperation() const override { return true; }
 
+  void addControls(const Controls& c) override {
+    for (auto ctrl : c) {
+      if (actsOn(ctrl.qubit)) {
+        throw QFRException(
+            "Cannot add control to operation as it already acts on "
+            "the control qubit.");
+      }
+
+      controls.insert(ctrl);
```

Comment:
Again, a minor detail that probably doesn't make the biggest difference here, but I believe the following should be preferred in almost any case
```suggestion
      controls.emplace(ctrl);
```